### PR TITLE
fix: correctly handle context overrides

### PR DESCRIPTION
**Pull request by theSuess**
The previous implementation was broken because of the way viper implements "Sub". The "overrides" section of the config only exists as env vars and can therefore not be used as a subconfig.

### CODE REVIEW

**Comment by malcolmholmes** [comment]
This is amazing. Only one comment before I think we can merge.

File: pkg/config/config.go [hunk 1]
```diff
@@ -128,21 +144,14 @@ func CreateContext(name string) error {
 }
 
 func Write() error {
```
[the remark this answers is not in the document]
Comment (theSuess) [reply pkg/config/config.go:146]: `Write()` only gets called as a result of either `Import` or `Set`. Only `Import` calls the `Override` function, and here it is intended that we take the overrides into account.

[the remark this answers is not in the document]
Comment (malcolmholmes) [reply pkg/config/config.go:146]: Override is also needed on use. If an envvar is set, it should "override" the context values. That's how viper works and I'd suggest what we should emulate.

[the remark this answers is not in the document]
Comment (theSuess) [reply pkg/config/config.go:146]: The override functionality is also used on `CurrentContext()` but that path does not lead to a call to `Write()`

[the remark this answers is not in the document]
Comment (malcolmholmes) [reply pkg/config/config.go:146]: Bingo, then I think we are Good.

**Review by malcolmholmes — APPROVED** [review]
I'll approve now. I can play more with this on Tuesday prior to thinking about a release.

**Pull request merged**